### PR TITLE
Initial pass on adding smoke tests for current versions of Firefox.

### PR DESCRIPTION
**Pull request by m8ttyB**
Replace the existing redirect tests for Firefox aliases (`test_redirect_for_firefox_aliases`) with one that is more deterministic.

An outline of what's included in this pr:
- create a utility class, `RelengHelper`, to pull down a json file containing the latest `{releng_alias: fx_version_num}`
- map releng aliases to go-bouncer aliases
- use the resulting mapping to have tests determine if go-bouncer is serving up the most recent versions of Firefox
- verify the releng aliases are what we expect them to be. This will give us a heads up if this list changes - `test_verify_releng_aliases_match_what_we_expect`.
- update the `cdn_netloc_locations` list to what is currently in use.
- remove the corresponding test from test_redirects.py, `test_redirect_for_firefox_aliases`.

/cc @stephendonner && @rpappalax && @rail

### CODE REVIEW

File: tests/e2e/tests/base.py [hunk 1]
```diff
@@ -10,9 +10,43 @@
...
+    cdn_netloc_locations = [
+        'download.cdn.mozilla.net',
+        'download-installer.cdn.mozilla.net'
+    ]
+
     _user_agent_firefox = ('Mozilla/5.0 (Macintosh; Intel Mac OS X 10.7; '
                            'rv:10.0.1) Gecko/20100101 Firefox/10.0.1')
 
+    def get_expected_fx_pkg_str(self, os, alias, product_version):
+        """Output a string reprentation of the expected Firefox file name.
```
Comment (stephendonner) [inline tests/e2e/tests/base.py]: Typo: "representation"
Nit: "filename"

**Comment by stephendonner** [comment]
Still working through my review, but it's starting to look pretty good!

(I'd happily file an Issue for this, but at some point we should consider adding in SHA-1 support, given https://download-sha1.allizom.org/ and its corresponding CDN https://download-sha1.cdn.mozilla.net/ are live.)

File: tests/e2e/tests/utils.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        for Firefox and version numbers as known by Mozilla's Release Engineering Team.
+
+        Release Engineering maintains an up-to-date JSON file with the current
+        Firefox release values - https://product-details.mozilla.org/1.0/firefox_versions.json.
+
+        :returns: {string:string} dictionary with releng aliases and version numbers"""
+        url = self._base_url % self._firefox_versions_uri
+        response = requests.get(url)
+        releng_products = response.json()
+
```
Comment (m8ttyB) [inline tests/e2e/tests/utils.py]: delete the empty space

**Comment by rbillings** [comment]
I reviewed the suggested changes, and they look good to me from a high level perspective. R+

**Comment by rpappalax** [comment]
agreeing you should throw an exception on line 47, otherwise r+

**Comment by m8ttyB** [comment]
Thank you all for the r+. Merging this madness.

**Pull request merged**